ocaml/backends/cpp.ml

```diff
@@ -1642,11 +1642,6 @@ let input_of_sim_package
                    | None -> None
                    | Some s -> Some (Util.string_of_coq_string s)); }
 
-let clang_format fname =
-  let style = "-style={BasedOnStyle: llvm, ColumnLimit: 100}" in
-  try Common.command "clang-format" ["-i"; style; fname]
-  with CompilationError _ -> ()
```

Comment:
Must we remove this?  It makes the generated cpp quite a bit nicer to look at.

Reply:
We do not have to, but I would strongly argue in favour of it. The dependency on `clang-format` should be optional, especially with regards to the size of a full LLVM installation that it pulls in as a dependency. It is easy enough to extend the dune rule to run `clang-format` if it is available. It should really not be part of `cuttlec`, because it is non of its concern.

Reply:
I see your point. My concern is this: it's a critical part of cuttlec that the output should be readable. When I built it I decided not to bother with pretty printing directly in cuttlec, because we could do that with an external tool; but the user is never supposed to see the unformatted code generated by cuttlec.  I forget whether CompilationError is the error that Common.command returns if the binary is missing. If not, we could catch that case too and print a warning, as a compromise?